test/middleware/jsonld.test.ts

```diff
@@ -1,105 +1,127 @@
+import { literal, namedNode, quad } from '@rdfjs/data-model';
 import { parse as parseContentType } from 'content-type';
-import { Context as JsonLdContext } from 'jsonld/jsonld-spec';
-import { Context, Response } from 'koa';
+import 'jest-rdf';
+import { addAll } from 'rdf-dataset-ext';
+import { Quad } from 'rdf-js';
+import { AppContext } from '../../src/app';
 import jsonld from '../../src/middleware/jsonld';
-import createContext from '../context';
-import runMiddleware, { Next } from '../middleware';
+import { rdf, schema } from '../../src/namespaces';
+import createContext, { Headers } from '../context';
+import { Next } from '../middleware';
 
-const makeRequest = async (next?: Next, jsonLdContext: JsonLdContext = {}): Promise<Response> => (
-  runMiddleware(jsonld(jsonLdContext), createContext(), next)
-);
+const makeRequest = async (
+  body?: string, headers?: Headers, next: Next = jest.fn(), jsonLdContext = {},
+): Promise<AppContext> => {
+  const context = createContext({ body, headers, method: body ? 'POST' : 'GET' });
+
+  await jsonld(jsonLdContext)(context, (): Promise<void> => next(context));
 
-const next = (body: unknown, type: string) => async ({ response }: Context): Promise<void> => {
-  response.body = body;
-  response.type = type;
+  return context;
 };
 
-const content = {
+const next = (body = null, quads: Array<Quad> = [], type = null, status = null) => (
+  async ({ response }: AppContext): Promise<void> => {
+    if (body) {
+      response.body = body;
+    }
+    if (quads) {
+      addAll(response.dataset, quads);
+    }
+    if (type) {
+      response.type = type;
+    }
+    if (status) {
+      response.status = status;
+    }
+  }
+);
+
+const id = namedNode('http://example.com/object');
+const quads = [
+  quad(id, rdf.type, schema.Article),
+  quad(id, schema('name'), literal('English Name', 'en')),
+  quad(id, schema('name'), literal('French Name', 'fr')),
+];
+
+const jsonLd = {
   '@id': 'http://example.com/object',
-  '@type': ['http://schema.org/Article'],
-  'http://schema.org/name': [{ '@value': 'English Name', '@language': 'en' }],
-  'http://purl.org/dc/elements/1.1/title': [
-    { '@value': 'English Title', '@language': 'en' },
-    { '@value': 'French Title', '@language': 'fr' },
-  ],
+  '@type': 'http://schema.org/Article',
+  'http://schema.org/name': [{ '@language': 'en', '@value': 'English Name' }, {
```

Comment:
that's some weird formatting for two values that should be equally important (`en`, `fr`)